GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -434,7 +434,7 @@ protected override void OnApplicationActivated()
         {
             if (!_bypassActivatedEventHandler && AppSettings.RefreshArtificialCommitOnApplicationActivated)
             {
-                RescanChanges();
+                RescanChangesAsync();
```

Comment:
The returned `Task` must be run, mustn't it?

Reply:
It shouldn't be awaited as it's fire-and-forget kind of task. I reverted this commit since I was returning this task because I didn't know about `ThreadHelper.JoinPendingOperations();` and that it does exactly what I need - waits for background tasks to complete (at least that's my assumption).